examples/8_record_dataset_from_gym.py

```diff
@@ -0,0 +1,266 @@
+
+        timestamps = []
+        drop_episode = False
+        for i in tqdm(range(num_steps)):
+            # Apply the next action
+            action = env.action_space.sample()
+            observation, _, _, _, info = env.step(action=action)
+
+            # store data
+            for key in observation:
+                obs_replay[key].append(copy.deepcopy(observation[key]))
+            obs_replay["action"].append(copy.deepcopy(action))
+
+            # Use simulator time, if available.
+            try:
+                timestamps.append(info["timestamp"])
+            except KeyError:
+                timestamps.append(np.float32(i)/args.fps)
+
+        print("stop")
+
+        if not drop_episode:
+            print(f"saving episode {ep_idx}")
+            ep_dict = {}
+
+            # store images in png and create the video
+            for img_key in args.image_keys.split(","):
+                save_images_concurrently(
+                    obs_replay[img_key],
+                    images_dir / f"{img_key}_episode_{ep_idx:06d}",
+                    args.num_workers,
+                )
+                fname = f"{img_key}_episode_{ep_idx:06d}.mp4"
+
+                # store the reference to the video frame
+                ep_dict[f"observation.{img_key}"] = [
+                    {"path": f"videos/{fname}", "timestamp": tstp} for tstp in timestamps]
+
+            states = []
+            for state_name in args.state_keys.split(","):
+                states.append(np.array(obs_replay[state_name]))
+            state = torch.tensor(np.concatenate(states, axis=1))
+
+            action = torch.tensor(np.array(obs_replay["action"]))
+            next_done = torch.zeros(num_steps, dtype=torch.bool)
+            next_done[-1] = True
+
+            ep_dict["observation.state"] = state
+            ep_dict["action"] = action
+            ep_dict["episode_index"] = torch.tensor(
+                [ep_idx] * num_steps, dtype=torch.int64)
+            ep_dict["frame_index"] = torch.arange(0, num_steps, 1)
+            ep_dict["timestamp"] = torch.tensor(timestamps)
+            ep_dict["next.done"] = next_done
+
+            print(f"num_steps={num_steps}")
+            print(f"timestamps[-1]={timestamps[-1]}")
+            ep_fps.append(num_steps / timestamps[-1])
+            ep_dicts.append(ep_dict)
+            print(f"Episode {ep_idx} done, fps: {ep_fps[-1]:.2f}")
+
+            episode_data_index["from"].append(id_from)
+            episode_data_index["to"].append(
+                id_from + num_steps if args.keep_last else id_from + num_steps - 1)
+
+            id_to = id_from + num_steps if args.keep_last else id_from + num_steps - 1
+            id_from = id_to
+
+            ep_idx += 1
+
+    env.close()
+
+    print("encode video frames")
+    for ep_idx in range(num_episodes):
+        for img_key in args.image_keys.split(","):
+            encode_video_frames(
+                vcodec="libx265",
```

Comment:
We should use the default `vcodec` (av1)

Reply:
We have an issue with ffmpeg av1 compatiblity on linux. You will need to run these:
```bash
conda install -c conda-forge ffmpeg
pip uninstall opencv-python
conda install -c conda-forge opencv>=4.10.0
```